sale_order_line_position/models/sale_order_line.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    position = fields.Integer(readonly=True, index=True, default=False)
+    position_formatted = fields.Char(compute="_compute_position_formatted")
+
+    @api.depends("position")
+    def _compute_position_formatted(self):
+        for record in self:
+            record.position_formatted = record._format_position(record.position)
+
+    @api.onchange("sequence")
+    def _onchange_sequence(self):
+        if self.order_id.locked_positions:
+            return
+        lines = self.order_id.order_line.filtered(
+            lambda x: not x.display_type and x.sequence < self.sequence
+        )
+        self.position = len(lines) + 1
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        sale_ids = [
+            line["order_id"]
+            for line in vals_list
+            if not line.get("display_type") and line.get("order_id")
+        ]
```

Comment:
If we just create a section, then sale_ids will be empty.
Therefore, the following query will raise an exception:
```
[...]
  File "/odoo/external-src/sale-reporting/sale_order_line_position/models/sale_order_line.py", line 40, in create
    self.env.cr.execute(query, (ids,))
  File "/odoo/src/odoo/sql_db.py", line 173, in wrapper
    return f(self, *args, **kwargs)
  File "/odoo/src/odoo/sql_db.py", line 250, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: syntax error at or near ")"
LINE 3:         WHERE order_id in () GROUP BY order_id; 
```
I'll push a fix for that.

Reply:
Thanks for your fix